pkg/domain/stateDiffComponent.go

```diff
@@ -0,0 +1,174 @@
+package domain
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	"golang.org/x/exp/maps"
+
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
+)
+
+// ComponentDiffs contains the differences for all expected Components to the current ecosystem.ComponentInstallations.
+type ComponentDiffs []ComponentDiff
+
+// Statistics aggregates various figures about the required actions of the ComponentDiffs.
+func (cd ComponentDiffs) Statistics() (toInstall int, toUpgrade int, toUninstall int, other int) {
+	for _, componentDiff := range cd {
+		switch componentDiff.NeededAction {
+		case ActionInstall:
+			toInstall += 1
+		case ActionUpgrade:
+			toUpgrade += 1
+		case ActionUninstall:
+			toUninstall += 1
+		default:
+			other += 1
+		}
+	}
+	return
+}
+
+// ComponentDiff represents the Diff for a single expected Component to the current ecosystem.ComponentInstallation.
+type ComponentDiff struct {
+	Name         string
+	Actual       ComponentDiffState
+	Expected     ComponentDiffState
+	NeededAction Action
+}
+
+// ComponentDiffState contains all fields to make a diff for components in respect to another ComponentDiffState.
+type ComponentDiffState struct {
+	Version           core.Version
```

Comment:
I do think so yes. Do you think this might be a problem for later stages of the blueprint-operator when component-operator and blueprint-operator version parsing may differ?

Reply:
Hm, I guess it won't be much of a problem. Even if it is, it can be easily changed since it's only in the domain.